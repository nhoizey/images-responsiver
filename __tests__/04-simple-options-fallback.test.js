'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('image with simple fallback option', () => {
	test('width attribute superior to the fallback', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test.png" width="789"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				fallbackWidth: 480,
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('width attribute inferior to the fallback', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test.png" width="420"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				fallbackWidth: 480,
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
