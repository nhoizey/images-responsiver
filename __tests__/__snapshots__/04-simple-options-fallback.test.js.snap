// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image with simple fallback option width attribute inferior to the fallback 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-420.png 420w\\"
      src=\\"test-420.png\\"
      width=\\"420\\"
    />
  </body>
</html>
"
`;

exports[`image with simple fallback option width attribute superior to the fallback 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-560.png 560w, test-789.png 789w\\"
      src=\\"test-480.png\\"
      width=\\"789\\"
    />
  </body>
</html>
"
`;
